numpy/core/src/_simd/_simd_inc_convert.h

```diff
@@ -0,0 +1,355 @@
+        if (!PyTuple_Check(obj) || PyTuple_GET_SIZE(obj) != tuple_len) {
+            PyErr_Format(PyExc_TypeError,
+                "a tuple of %d vector type %s is required",
+                tuple_len, simd_data_getinfo(vdtype)->pyname
+            );
+            return data;
+        }
+        for (int i = 0; i < tuple_len; ++i) {
+            PyObject *item = PyTuple_GET_ITEM(obj, i);
+            if (item == NULL) {
+                PyErr_Format(PyExc_RuntimeError,
+                    "unable to get item %d, expected to get a %s",
+                    i, simd_data_getinfo(vdtype)->pyname
+                );
+                return data;
+            }
+            // let the compiler casting for us
+            if (i == 2) {
+                data.vu64x3.val[2] = simd_vector2data((simd_vector*)item, dtype).vu64;
+            } else {
+                data.vu64x2.val[i] = simd_vector2data((simd_vector*)item, dtype).vu64;
+            }
+            // TODO: improve log add item number
+            if (PyErr_Occurred()) {
+                return data;
+            }
+        }
+        return data;
+    }
+    PyErr_Format(PyExc_RuntimeError,
+        "unhandled obj2data type id:%d, name:%s",
+        dtype, simd_data_getinfo(dtype)->pyname
+    );
+    return data;
+}
+
+static PyObject *
+simd_data2obj(simd_data data, simd_data_type dtype)
+{
+    if (simd_data_is_scalar(dtype)) {
+        return simd__data2obj_scalar(data, dtype);
+    }
+    if (simd_data_is_sequence(dtype)) {
+        int lane_size = simd_data_getinfo(dtype)->lane_size;
+        Py_ssize_t dseq_size = simd__alloc_size(data.qu8) / lane_size;
+        PyObject *list = PyList_New(dseq_size);
+        if (list == NULL) {
+            return NULL;
+        }
+        if (!simd_data2seq(list, data, dtype)) {
+            Py_DECREF(list);
+            return NULL;
+        }
+        return list;
+    }
+    if (simd_data_is_vector(dtype)) {
+        return (PyObject*)simd_data2vector(data, dtype);
+    }
+    int is_x2 = simd_data_is_vectorx2(dtype);
+    int is_x3 = simd_data_is_vectorx3(dtype);
+    if (is_x2 || is_x3) {
+        int tuple_len = is_x3 ? 3 : 2;
+        PyObject *tuple = PyTuple_New(tuple_len);
+        if (tuple == NULL) {
+            return NULL;
+        }
+        simd_data_type vdtype = simd_data_cvt2vector(dtype);
+        for (int i = 0; i < tuple_len; ++i) {
+            simd_data vdata;
+            if (i == 2) {
+                vdata.vu64 = data.vu64x3.val[2];
+            } else {
+                vdata.vu64 = data.vu64x2.val[i];
+            }
+            PyObject *item = (PyObject*)simd_data2vector(vdata, vdtype);
+            if (item == NULL) {
+                // TODO: improve log add item number
+                Py_DECREF(tuple);
+                return NULL;
+            }
+            PyTuple_SET_ITEM(tuple, i, item);
+        }
+        return tuple;
+    }
+    /**end repeat**/
```

Comment:
What is this ending? I don't see the start

Reply:
forget to remove it during the latest cleanup, also `_simd_inc_convert.h ` became a normal header not a template anymore.